sklearn/preprocessing/tests/test_data.py

```diff
@@ -1952,6 +1955,98 @@ def test_one_hot_encoder_unknown_transform():
     assert_raises(ValueError, oh.transform, y)
 
 
+def check_categorical(X):
+    enc = CategoricalEncoder()
+    Xtr1 = enc.fit_transform(X)
+
+    enc = CategoricalEncoder()
+    Xtr2 = enc.fit_transform(X)
+
+    enc = CategoricalEncoder(sparse=False)
+    Xtr3 = enc.fit_transform(X)
+
+    assert_allclose(Xtr1.toarray(), Xtr2.toarray())
```

Comment:
good catch, before they did something different (different selection of the columns, but i removed that feature), so not needed anymore